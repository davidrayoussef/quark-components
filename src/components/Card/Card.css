.card {
  display: flex;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  text-align: center;
  background: #f2f2f2;
  max-width: 250px;
  padding: 30px;
}

.card img {
  height: 100px;
}

.card h3 {
  margin: 0 0 10px 0;
  font-size: 1.5em;
  font-weight: 400;
}

.description {
  margin-bottom: 10px;
}
