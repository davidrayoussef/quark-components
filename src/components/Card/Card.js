import React from 'react';
import PropTypes from 'prop-types';

const Card = ({ name, avatar_url, description, html_url }) => {
  return (
    <div>
      {avatar_url && <div><img src={avatar_url} alt={name} style={{height: 100}}/></div>}
      <div>{name}</div>
      {description && <div>{description}</div>}
      {html_url && <div><a href={html_url}>{html_url}</a></div>}
    </div>
  );
};

Card.proptypes = {
  name: PropTypes.string.isRequired,
  avatar_url: PropTypes.string,
  location: PropTypes.string,
  html_url: PropTypes.string
};

export default Card;
