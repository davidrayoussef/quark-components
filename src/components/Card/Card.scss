.card {
  display: flex;
  gap: 0.5rem;
  flex-direction: column;
  justify-items: center;
  align-items: center;
  text-align: center;
  background: #f2f2f2;
  max-width: 250px;
  padding: 30px;
  box-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
  & h3 {
    margin: -5px 0 10px;
    font-size: 1.5em;
    font-weight: 400;
  }
  & img {
    height: 100px;
  }
}
