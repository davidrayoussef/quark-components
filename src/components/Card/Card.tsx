import React from 'react';

import { CardProps } from '../../shared';

import style from './Card.css';

export const Card: React.FC<CardProps> = ({
  children,
  title,
  ...rest
}: CardProps) => {
  return (
    <div className={style.card} {...rest}>
      {title && <h3>{title}</h3>}
      {children}
    </div>
  );
};
