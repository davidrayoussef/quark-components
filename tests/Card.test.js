import React from 'react';
import { shallow } from 'enzyme';
import Card from '../src/components/Card/Card';

describe('<Card />', () => {
  it('should render', () => {
    const wrapper = shallow(<Card />);
    expect(wrapper);
  });
});
