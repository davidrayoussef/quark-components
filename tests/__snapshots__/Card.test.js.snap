// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  className="card"
>
  <h3>
    Title
  </h3>
</div>
`;
